helix-term/src/commands.rs

```diff
@@ -3571,6 +3571,16 @@ pub fn completion(cx: &mut Context) {
         None => return,
     };
 
+    let capabilities = language_server.capabilities();
+    let mut trigger_list = Vec::new();
+    if let Some(lsp::CompletionOptions {
+        trigger_characters: Some(triggers),
+        ..
+    }) = &capabilities.completion_provider
+    {
+        trigger_list = triggers.clone();
+    }
```

Comment:
```rust
let trigger_list = language_server
    .capabilities()
    .completion_provider
    .and_then(|options| options.trigger_characters)
    .unwrap_or(Vec::new());
```

Reply:
@the-mikedavis Thanks! That looks a lot better! 
The compiler told me to add some stuff so I did 
```rust
    let trigger_list = language_server
    .capabilities()
    .completion_provider.as_ref()
    .and_then(|options| options.trigger_characters.as_ref())
    .unwrap_or(Vec::new());
    ```
    Now the error is 
 ```sh
    error[E0308]: mismatched types
    --> helix-term/src/commands.rs:3578:16
     |
3578 |     .unwrap_or(Vec::new());
     |                ^^^^^^^^^^
     |                |
     |                expected reference, found struct `Vec`
     |                help: consider borrowing here: `&Vec::new()`
     |
     = note: expected reference `&Vec<std::string::String>`
                   found struct `Vec<_>`
```
I tried `&Vec::new()`
```sh
error[E0716]: temporary value dropped while borrowed
    --> helix-term/src/commands.rs:3578:17
     |
3578 |     .unwrap_or(&Vec::new());
     |                 ^^^^^^^^^^ - temporary value is freed at the end of this statement
     |                 |
     |                 creates a temporary which is freed while still in use
...
3597 |     let offset = if trigger_list.is_empty() {
     |                     ----------------------- borrow later used here
     |
     = note: consider using a `let` binding to create a longer lived value
```

Reply:
Ah sorry I had some borrowing mistakes 😅. I edited the comment to add a `clone` in the middle that should fix it